DiceKitTests/SubtractionExpressionResult_Tests.swift

```diff
@@ -0,0 +1,125 @@
+//
+//  SubtractionExpressionResult_Tests.swift
+//  DiceKit
+//
+//  Created by Jonathan Hoffman on 8/15/15.
+//  Copyright © 2015 Brentley Jones. All rights reserved.
+//
+
+import XCTest
+import Nimble
+import SwiftCheck
+
+import DiceKit
+
+/// Tests the `SubtractionExpressionResult` type
+class SubtractionExpressionResult_Tests: XCTestCase {
+    
+}
+
+// MARK: - Equatable
+extension SubtractionExpressionResult_Tests {
+    
+    func test_shouldBeReflexive() {
+        property("reflexive") <- forAll {
+            (a: Constant, b: Constant) in
+            
+            return EquatableTestUtilities.checkReflexive { SubtractionExpressionResult(a, b) }
+        }
+    }
+    
+    func test_shouldBeSymmetric() {
+        property("symmetric") <- forAll {
+            (a: Constant, b: Constant) in
+            
+            return EquatableTestUtilities.checkSymmetric { SubtractionExpressionResult(a, b) }
+        }
+    }
+    
+    func test_shouldBeTransitive() {
+        property("transitive") <- forAll {
+            (a: Constant, b: Constant) in
+            
+            return EquatableTestUtilities.checkTransitive { SubtractionExpressionResult(a, b) }
+        }
+    }
+    
+    func test_shouldBeAbleToNotEquate() {
+        property("non-equal") <- forAll {
+            (a: Constant, b: Constant, m: Constant, n: Constant) in
+            
+            return (!(a == m && b == n) && !(a == n && b == m)) ==> {
+                EquatableTestUtilities.checkNotEquate(
+                    { SubtractionExpressionResult(a, b) },
+                    { SubtractionExpressionResult(m, n) }
+                )
+            }
+        }
+    }
+    
+    func test_shouldBeAnticommutative() {
+        property("Anticommutative") <- forAll {
+            (a: Constant, b: Constant) in
+            
+            return (a != b) ==> {
+                let x = SubtractionExpression(a, b).evaluate()
+                let y = SubtractionExpression(b, a).evaluate()
+                
+                return x.value == -y.value
```

Comment:
Sure, since that would check the equality operation on the result type, when here I am technically testing a mathematical property of the values they return.